lib/src/tes/mod.rs

```diff
@@ -0,0 +1,328 @@
+        let instance = TES {
+            config: config.clone(),
+            transport,
+            service: resp,
+        };
+
+        if instance.check() {
+            Ok(instance)
+        } else {
+            Err("The endpoint is not an instance of TES".into())
+        }
+    }
+
+    fn check(&self) -> bool {
+        let resp = &self.service;
+        match resp.as_ref() {
+            Ok(service) => service.r#type.artifact == "tes",
+            Err(_) => false, // or handle the error as appropriate
+        }
+    }
+
+    pub async fn create(
+        &self,
+        task: TesTask, /*, params: models::TesTask*/
+    ) -> Result<Task, Box<dyn std::error::Error>> {
+        // First, check if the service is of TES class
+        if !self.check() {
+            // If check fails, log an error and return an Err immediately
+            log::error!("Service check failed");
+            return Err("Service check failed".into());
+        }
+        // todo: version in url based on serviceinfo or user config
+        let response = self
+            .transport
+            .post("/ga4gh/tes/v1/tasks", Some(json!(task)))
+            .await;
+        match response {
+            Ok(response_body) => {
+                let v: serde_json::Value = serde_json::from_str(&response_body)?;
+
+                // Access the `id` field
+                let task_id = v
+                    .get("id")
+                    .and_then(|v| v.as_str())
+                    .unwrap_or_default()
+                    .trim_matches('"')
+                    .to_string();
+
+                let task = Task {
+                    id: task_id,
+                    transport: self.transport.clone(),
+                };
+                Ok(task)
+            }
+            Err(e) => Err(Box::new(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                format!("Failed to post task: {}", e),
+            ))),
+        }
+    }
+
+    pub async fn get(&self, view: &str, id: &str) -> Result<TesTask, Box<dyn std::error::Error>> {
+        let task_id = id;
+        let url = format!("/tasks/{}?view={}", task_id, view);
+        // let params = [("view", view)];
+        // let params_value = serde_json::json!(params);
+        // let response = self.transport.get(&url, Some(params_value)).await;
+        let response = self.transport.get(&url, None).await;
+        match response {
+            Ok(resp_str) => {
+                let task: TesTask = from_str(&resp_str)?;
+                Ok(task)
+            }
+            Err(e) => Err(e),
+        }
+    }
+    pub async fn list_tasks(
+        &self,
+        params: Option<ListTasksParams>,
+    ) -> Result<TesListTasksResponse, Box<dyn std::error::Error>> {
+        let params_value = params.map(|p| {
+            let mut map = serde_json::Map::new();
+            
+            if let Some(name_prefix) = p.name_prefix {
```

Comment:
Yeah, I agree. Do you think we should create an issue and leave it for now, or try resolving it now?

Reply:
I think we can keep post processing of params as it is, as earlier I was entering the params in a wrong manner, which caused such long and complex workarounds. I have changed it now and it now takes only 1 line instead of checking NULL values for each attribute in the param, using the function:
```
fn serialize_to_json<T: Serialize>(item: T) -> Value {
    serde_json::to_value(&item).unwrap()
}
```
and adding
```
#[serde(skip_serializing_if = "Option::is_none")]
```
to the ListTasksParams struct